pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
+
+	ImportPolicy    TagImportPolicy `json:"importPolicy,omitempty" description:"policy controlling how the image is imported"`
+	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
+
+	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
```

Comment:
> the user's desired intent SHOULD have a "spec" and a "status".  Objects
> whose state cannot vary from the user's desired intent MAY have only
> "spec", and MAY rename "spec" to a more appropriate name.

I would argue that in the case of an actual import, what you've describe is exactly the case for spec and status.  The user uses `spec` to describe his desired state: "I want these things imported" and the API server uses `status` to indicate whether or not it was done.